### PR TITLE
[IMP] contract: Module contract improvements

### PR DESCRIPTION
**Pull request by carlosdauden**
Improve module adding date_end, filters and partner smartbutton

@Tecnativa

### CODE REVIEW

**Comment by sergio-teruel** [comment]
Code review LGTM :+1: 
Please check travis

File: contract/i18n/es.po [hunk 1]
```diff
@@ -1,28 +1,30 @@
...
+"Content-Transfer-Encoding: 8bit\n"
 "Plural-Forms: nplurals=2; plural=(n != 1);\n"
+"X-Generator: Poedit 1.8.7.1\n"
 
 #. module: contract
 #: model:mail.template,body_html:contract.email_contract_template
 msgid ""
 "\n"
-"<div style=\"font-family: 'Lucida Grande', Ubuntu, Arial, Verdana, sans-serif; font-size: 12px; color: rgb(34, 34, 34); background-color: #FFF; \">\n"
+"<div style=\"font-family: 'Lucida Grande', Ubuntu, Arial, Verdana, sans-"
```
Comment (pedrobaeza) [inline contract/i18n/es.po]: Please don't edit this with POedit or remove the line wrapping option or you will end up with this situation of a lot of lines diff without reason

**Comment by carlosdauden** [comment]
@pedrobaeza changes done

**Comment by rafaelbn** [comment]
@carlosdauden travis is red. Please could you review? Thanks!

**Review by rafaelbn — APPROVED** [review]
Tested functionally in runbot 👍

File: contract/models/res_partner.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    contract_count = fields.Integer(
+        compute='_compute_contract_count',
+        string='Contracts')
+
+    def _compute_contract_count(self):
+        Contract = self.env['account.analytic.account']
+        today = fields.Date.today()
+        for partner in self:
+            partner.contract_count = Contract.search_count([
+                ('recurring_invoices', '=', True),
```
Comment (chienandalu) [inline contract/models/res_partner.py:21]: Is it needed to filter by this criteria?

**Review by chienandalu — COMMENTED** [review]
Shouldn't README.rst be updated

File: contract/models/res_partner.py (hunk 2 above)
Comment (carlosdauden) [reply contract/models/res_partner.py:21]: It seems that you still do not know
that users are capable :wink: 

What is the problem?

Comment (chienandalu) [reply contract/models/res_partner.py:21]: No, it's just my ignorance on the deeper concepts of the module :smile: Can't a contract with no invoices be considered valid?

**Comment by chienandalu** [comment]
By the way, travis has the same issue as here https://github.com/OCA/contract/pull/104 and here https://github.com/OCA/partner-contact/pull/470 (built gets stalled)

File: contract/models/res_partner.py (hunk 2 above)
Comment (carlosdauden) [reply contract/models/res_partner.py:21]: https://github.com/OCA/contract/pull/107/files#diff-47b79491db78ebebcb372014ce9c7bf3R39

File: contract/models/account_analytic_account.py [hunk 3]
```diff
@@ -29,9 +29,15 @@ class AccountAnalyticAccount(models.Model):
...
-    date_start = fields.Date(default=fields.Date.context_today)
+    date_start = fields.Date(
+        default=fields.Date.context_today,
+        string='Date Start')
+    date_end = fields.Date(
+        index=True,
+        string='Date End')
     recurring_invoices = fields.Boolean(
         string='Generate recurring invoices automatically',
+        index=True,
```
Comment (pedrobaeza) [inline contract/models/account_analytic_account.py]: An index over a boolean is useless

File: contract/views/account_analytic_account_view.xml [hunk 4]
```diff
@@ -104,15 +105,29 @@
...
                 <field name="journal_id"/>
                 <field name="pricelist_id"/>
+                <separator/>
                 <filter name="recurring_invoices"
                         string="Recurring Invoices"
                         domain="[('recurring_invoices','=',True)]"
                         />
+                <separator/>
+                <filter name="not_finished"
+                        string="Not Finished"
```
Comment (pedrobaeza) [inline contract/views/account_analytic_account_view.xml]: You can put `Valid` as the term for "Vigente".

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by carlosdauden** [comment]
@pedrobaeza changes done

File: contract/models/account_analytic_account.py [hunk 5]
```diff
@@ -29,7 +29,12 @@ class AccountAnalyticAccount(models.Model):
         inverse_name='analytic_account_id',
         copy=True,
     )
-    date_start = fields.Date(default=fields.Date.context_today)
+    date_start = fields.Date(
+        default=fields.Date.context_today,
+        string='Date Start')
```
Comment (lasley) [inline contract/models/account_analytic_account.py]: [Please add a trailing comma and put the `)` on next line](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#idioms)

[More info - note about version control](https://www.python.org/dev/peps/pep-0008/#when-to-use-trailing-commas)

File: contract/models/account_analytic_account.py [hunk 6]
```diff
@@ -29,7 +29,12 @@ class AccountAnalyticAccount(models.Model):
...
+    date_end = fields.Date(
+        index=True,
+        string='Date End')
```
Comment (lasley) [inline contract/models/account_analytic_account.py]: And here

File: contract/models/res_partner.py (hunk 2 above)
Comment (lasley) [reply contract/models/res_partner.py:21]: It's just ambiguous (legacy) naming

File: contract/models/res_partner.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        self.ensure_one()
+        res = self.env['ir.actions.act_window'].for_xml_id(
+            'contract', 'action_account_analytic_overdue_all')
+        res.update(
+            context=dict(
+                self.env.context,
+                search_default_recurring_invoices=True,
+                search_default_not_finished=True,
+                default_partner_id=self.id,
+                default_recurring_invoices=True),
```
Comment (lasley) [inline contract/models/res_partner.py]: Add trailing comma and put `)` on next line please.

File: contract/models/res_partner.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    contract_count = fields.Integer(
+        compute='_compute_contract_count',
+        string='Contracts')
```
Comment (lasley) [inline contract/models/res_partner.py]: Trailing comma

File: contract/models/res_partner.py [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+        res = self.env['ir.actions.act_window'].for_xml_id(
+            'contract', 'action_account_analytic_overdue_all')
+        res.update(
+            context=dict(
+                self.env.context,
+                search_default_recurring_invoices=True,
+                search_default_not_finished=True,
+                default_partner_id=self.id,
+                default_recurring_invoices=True),
+            domain=[('partner_id', '=', self.id)])
```
Comment (lasley) [inline contract/models/res_partner.py]: and here

**Review by lasley — CHANGES_REQUESTED** [review]
Thanks @carlosdauden. It would also be nice to see tests making sure that the end date works and the contract count is correct (a new test for every feature).

File: contract/models/res_partner.py (hunk 2 above)
Comment (carlosdauden) [reply contract/models/res_partner.py:21]: I agree.

https://github.com/odoo/odoo/blob/6140300da011d9eb4196b780b011bbc87fe8cae0/addons/account_analytic_analysis/account_analytic_analysis.py#L528
https://github.com/OCA/contract/blob/35d6358471c071ecdb17d936c0b5e11ee28a86ee/contract/models/contract.py#L114

**Comment by carlosdauden** [comment]
@lasley Thanks for you review.
Changes done.

**Review by lasley — APPROVED** [review]
Changes look good, thanks @carlosdauden. Tentative approval from me pending a 🍏  CI

**Comment by rafaelbn** [comment]
@pedrobaeza could we go ahead with this one please?

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Travis error unrelated. Merging.

**Pull request merged**